apps/comps/app/competitions/[id]/og-image/route.tsx

```diff
@@ -156,12 +220,532 @@ async function loadAssets(): Promise<CachedAssets> {
+
+  return (
+    <div
+      tw="flex flex-col w-full rounded-lg overflow-hidden"
+      style={{ border: `1px solid ${COLORS.table.border}` }}
+    >
+      {/* Header */}
+      <div
+        tw="flex w-full px-8 py-3 text-sm font-semibold tracking-wider justify-between"
+        style={{
+          backgroundColor: COLORS.table.headerBg,
+          color: COLORS.table.headerText,
+          borderBottom: `1px solid ${COLORS.table.border}`,
+        }}
+      >
+        <div tw="w-20">Rank</div>
+        <div tw="flex-1">Agent</div>
+        <div>{valueColumnHeader}</div>
+      </div>
+
+      {/* Rows */}
+      {entries.map((entry, index) => (
+        <LeaderboardRow
+          key={entry.id}
+          entry={entry}
+          index={index}
+          isLast={index === entries.length - 1}
+        />
+      ))}
+    </div>
+  );
+}
+
+/**
+ * Props for the CTA button component.
+ */
+interface CtaButtonProps {
+  text: string;
+}
+
+/**
+ * CTA button component for the OG image.
+ * @param props - Component props.
+ * @param props.text - The button text.
+ * @returns React element for the CTA button.
+ */
+function CtaButton({ text }: CtaButtonProps): ReactElement {
+  return (
+    <div
+      tw="flex items-center justify-center rounded-xl text-white font-semibold mt-20 px-42 py-7 text-4xl tracking-widest w-full"
+      style={{
+        background: `linear-gradient(180deg, ${COLORS.button.light} 0%, ${COLORS.button.base} 100%)`,
+        boxShadow: "0 4px 20px rgba(0, 0, 0, 0.45)",
+        color: COLORS.text.light,
+      }}
+    >
+      {text}
+    </div>
+  );
+}
+
+/**
+ * Props for the right section active component.
+ */
+interface RightSectionActiveOrEndedProps {
+  competition: Competition;
+  entries: LeaderboardEntry[];
+}
+
+/**
+ * Right section active component for the OG image.
+ * @param props - Component props.
+ * @param props.competition - The competition object.
+ * @param props.entries - Array of leaderboard entries.
+ * @returns React element for the right section when competition is active or ended.
+ */
+function RightSectionActiveOrEnded({
+  competition,
+  entries,
+}: RightSectionActiveOrEndedProps): ReactElement {
+  return (
+    <div tw="flex flex-col pt-30 w-full text-white h-full justify-between">
+      <LeaderboardTable entries={entries} competitionType={competition.type} />
+      <CtaButton text={getCtaText(competition)} />
+    </div>
+  );
+}
+
+/**
+ * Props for the right section pending component.
+ */
+interface RightSectionPendingProps {
+  competition: Competition;
+  totalRewards: bigint;
+  tokenSvg: string;
+}
+
+/**
+ * Right section pending component for the OG image.
+ * @param props - Component props.
+ * @param props.competition - The competition object.
+ * @param props.totalRewards - The total rewards amount.
+ * @param props.tokenSvg - The Recall token SVG data URL.
+ * @returns React element for the right section when competition is pending.
+ */
+function RightSectionPending({
```

Comment:
aka when a comp is pending, the right hand side will always show the total reward pool, and not the agents.